src/WebJobs.Script.WebHost/Management/LinuxSpecialization/RunFromPackageCloudBlockBlobService.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Management.LinuxSpecialization
+{
+    public class RunFromPackageCloudBlockBlobService
+    {
+        public virtual async Task<bool> BlobExists(string url, string environmentVariableName, ILogger logger)
+        {
+            return await BlobExistsAsync(url, environmentVariableName, logger);
+        }
+
+        private static async Task<bool> BlobExistsAsync(string url, string environmentVariableName, ILogger logger)
```

Comment:
yes thats the idea, but seemed like an interface was not necessary since we can achieve this using virtual method. So leaning towards leaving this as is.

Reply:
I think both are valid approaches and following Mathew's feedback from an earlier PR since I already have it implemented this way  :)

https://github.com/Azure/azure-functions-host/pull/2962#discussion_r194864519
Note that you can mock using virtuals, without defining interfaces. I.e. the IHostAssignmentContextReader/ILinuxContainerInitializationService/etc. interfaces aren't really necessary. Interfaces are useful if you expect to have different implementations in the future, but that isn't the case for any of these.

Reply:
thanks for explanation.